CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ option(PRODUCT_EMBED_BUILD_INFO "Embed build revision information into plProduct
 cmake_dependent_option(PRODUCT_EMBED_BUILD_TIME "Embed build time into plProduct" ON [[PRODUCT_EMBED_BUILD_INFO]] OFF)
 
 # HeadSpin Configuration
+
+# Define HS_DEBUGGING for debug builds
+add_compile_definitions($<$<CONFIG:Debug>:HS_DEBUGGING>)
```

Comment:
Ideally we would be moving away from these scoped commands and toward target based stuff, but there's already a precedent for this below, so I don't think this is a blocker.